### PR TITLE
[receiver/dockerstats] Disable container.cpu.usage.system by default, and expand description

### PR DESCRIPTION
**Pull request by jamesmoessis**
**Description:**
* `container.cpu.usage.system` is no longer a default metric. 
* Added more description for the metric, since I believe it had the possibility to mislead users. 

My rationale behind disabling this as a default metric, is that it has the potential to be misleading unless the user has explicitly read the description. It also duplicates system CPU metrics emitted by the `hostmetricsreceiver`. I thought about completely removing it, but I thought I better leave it in since it is something that docker reports. 

Change is technically breaking, but only behind the non-default featuregate. I don't expect it will impact any users. 

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9794

**Testing:** 

Tests were updated to test for the updated description

**Documentation:**

Go generate / mdatagen updates the docs.

### CODE REVIEW

**Comment by jamesmoessis** [comment]
FYI @MovieStoreGuy @rmfitzpatrick

File: receiver/dockerstatsreceiver/metadata.yaml [hunk 1]
```diff
@@ -40,8 +40,9 @@ attributes:
 metrics:
   # CPU
   container.cpu.usage.system:
```
Comment (bogdandrutu) [inline receiver/dockerstatsreceiver/metadata.yaml:42]: All the cpu metrics seem to me inconsistent with specification and other cpu usage metrics. We better take the opportunity on fixing all of them now.

Comment (jamesmoessis) [reply receiver/dockerstatsreceiver/metadata.yaml:42]: @bogdandrutu can you clarify in what ways the CPU metrics are inconsistent with the spec?

The CPU metrics here are taken directly from what the Docker API provides - here is the [exact struct](https://github.com/moby/moby/blob/master/api/types/stats.go#L7-L55) we pull from.

The [semconv](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/system-metrics.md#systemcpu---processor-metrics) doesn't define how to report cgroup/container level metrics. This receiver only reports cgroup/container metrics, with the exception of the system-level `container.cpu.usage.system` which comes from the Docker API. I believe its main purpose is to compare the system's overall usage with a container's usage, which can be done via `100*container.cpu.usage.total` / `container.cpu.usage.system` = container's CPU usage as a percentage of the system's. 

The semconv only defines system-level CPU metrics `system.cpu.time` and `system.cpu.utilization`. These don't align 100% with what the docker API provides. The only clash/duplicate is `container.cpu.usage.system` with the non-idle states of `system.cpu.time` - I'd be happy to remove `container.cpu.usage.system` from this receiver if you believe it is clashing with the semconv/spec/hostmetricsreceiver.

Comment (jamesmoessis) [reply receiver/dockerstatsreceiver/metadata.yaml:42]: @bogdandrutu any thoughts? I would like to get this PR moving.

**Review by bogdandrutu — COMMENTED** [review]
@open-telemetry/collector-contrib-approvers how do you want to do this?

**Comment by MovieStoreGuy** [comment]
@bogdandrutu , not an approver but as I understand that this change is a low risk due to the fact that receiver has yet to adopt the version 2 of the scraper as the default. (Version 2 captures metrics using the generated metrics file)

**Comment by mx-psi** [comment]
IMHO since the stability of this feature gate is alpha https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/f9e2bcb214ddb8dddf2ffbfb7d69a3fbc440b404/receiver/dockerstatsreceiver/factory.go#L32 we just need to list the change as breaking and move on. The whole point of feature gates is to allow doing breaking changes more quickly, we should make clear that this is breaking but that's about it.

**Comment by dmitryax** [comment]
If it's behind a feature gate that is disabled by default I believe we can move on

**Review by dmitryax — APPROVED** [review]

**Pull request merged**